product_variant_exclusion/views/product.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="product_template_only_form_view" model="ir.ui.view">
+        <field name="name">product.template.product.form</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_only_form_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="//page[@name='variants']//field[@name='attribute_line_ids']" position="after">
+                <group>
+                    <field name="allowed_value_ids" invisible="1"/>
+                    <field name="variant_exclusion_ids" widget="one2many_list" context="{'show_attribute': True}" nolabel="1">
+                        <tree string="Exclusions" editable="bottom">
```

Comment:
Given that this is positioned under the attribute values, consider using top for editable, that way user can keep a view of attributes onscreen regardless of how long the exclusions get.